stubs/openpyxl/openpyxl/reader/excel.pyi

```diff
@@ -1,16 +1,25 @@
-from _typeshed import Incomplete
+from _typeshed import Incomplete, StrPath
+from typing import IO
+from typing_extensions import Final, Literal
+from zipfile import ZipFile
 
-SUPPORTED_FORMATS: Incomplete
+from openpyxl.packaging.relationship import Relationship
+from openpyxl.packaging.workbook import ChildSheet
+from openpyxl.workbook.workbook import Workbook
+
+SUPPORTED_FORMATS: Final[tuple[Literal[".xlsx"], Literal[".xlsm"], Literal[".xltx"], Literal[".xltm"]]]
 
 class ExcelReader:
-    archive: Incomplete
-    valid_files: Incomplete
-    read_only: Incomplete
-    keep_vba: Incomplete
-    data_only: Incomplete
-    keep_links: Incomplete
-    shared_strings: Incomplete
-    def __init__(self, fn, read_only: bool = ..., keep_vba=..., data_only: bool = ..., keep_links: bool = ...) -> None: ...
+    archive: ZipFile
+    valid_files: list[str]
+    read_only: bool
+    keep_vba: bool
+    data_only: bool
+    keep_links: bool
+    shared_strings: list[str]
+    def __init__(
+        self, fn: StrPath | IO[bytes], read_only: bool = ..., keep_vba: bool = ..., data_only: bool = ..., keep_links: bool = ...
```

Comment:
Conflict:
This, and `load_workbook` is ultimately passed to `ZipFile` in `_validate_archive`.
`ZipFile.__init__` is typed as `ZipFile(file: StrPath | IO[bytes], ...)`

Even in read mode it still needs a `name` variable, `seek` and `tell` methods.